aes-kw/src/kwp.rs

```diff
@@ -1,11 +1,17 @@
-use crate::{ctx::Ctx, Error, IV_LEN, SEMIBLOCK_SIZE};
+use core::ops::{Add, Div, Mul};
+
+use crate::{ctx::Ctx, Error, IntegrityCheckFailed, IvLen, IV_LEN};
 use aes::cipher::{
+    array::ArraySize,
+    consts::{B1, U4294967296, U7},
     crypto_common::{InnerInit, InnerUser},
-    typenum::U16,
-    Block, BlockCipherDecrypt, BlockCipherEncrypt,
+    typenum::{Add1, IsLess, Le, NonZero, Prod, Quot, Sum, U16},
+    Array, Block, BlockCipherDecrypt, BlockCipherEncrypt,
 };
 
-/// Maximum length of the AES-KWP input data (2^32 bytes).
+/// Maximum length of the AES-KWP input data (2^32 bytes) represented as a `typenum` type.
+type KwpMaxLen = U4294967296;
```

Comment:
```suggestion
/// Maximum length of the AES-KWP input data (2^32 bytes - 1) represented as a `typenum` type.
type KwpMaxLen = U4294967295;
```

And use `LeEq` when consumed?

Reply:
`typenum` does not provide an alias for `U4294967295`. We would have to define it ourselves, but the resulting code will be less clear.

Reply:
ha, disregard then.